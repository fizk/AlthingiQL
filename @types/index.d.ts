export interface AssemblySummary {
    id: number,
    period: {
        from?: string,
        to?: string
    },
    division: {
        majority?: {
            id?: number,
            name?: string,
            color?: string
        }[],
        minority: {
            id?: number,
            name?: string,
            color?: string
        }[]
    }
}

export interface Picture {
    templateSrc: string
}

export interface Party {
    id: number
    name: string
    color: string
}

export interface Congressman {
    id: number
    name: string
    avatar: Picture
    party: Party
}

export interface DocumentVote {
    id: number
    voteId: number
    vote: string
    congressman: Congressman
}

export interface AssemblyIssue {
    id: number
    assembly?: {id: number}
    category?: string
    name: string
    subName?: string
    type?: string
    typeName: string
    typeSubName?: string
    status: string
    question?: string
    goal?: string
    date: string
}

export interface AssemblyCategorySummary {
    id: number
    superCategoryId: number
    title: string
    count: number
}


declare var __IMAGE_SERVER__: string;
declare var __GRAPHQL_SERVER__: string;

// declare const __IMAGE_SERVER__: string;
// declare const __GRAPHQL_SERVER__: string;
// declare const __APOLLO_STATE__: string;

declare global {
    interface Window {
        __APOLLO_STATE__: any
    }
}
