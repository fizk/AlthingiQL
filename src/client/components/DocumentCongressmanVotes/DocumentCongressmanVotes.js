import React from 'react';
import PropTypes from 'prop-types';
import {Congressman} from "../../elements/Congressman/index";

export default class DocumentCongressmanVotes extends React.Component {
    static propTypes = {
        assembly: PropTypes.number,
        issue: PropTypes.number,
        vote: PropTypes.number,
        votes: PropTypes.arrayOf(PropTypes.shape({
            id: PropTypes.number,
            voteId: PropTypes.number,
            vote: PropTypes.string,
            congressman: PropTypes.shape({
                id: PropTypes.number,
                name: PropTypes.string,
                party: PropTypes.shape({
                    id: PropTypes.number,
                    name: PropTypes.string,
                    color: PropTypes.string
                }),
            }),
        })),
    };

    static defaultProps = {
        assembly: undefined,
        issue: undefined,
        vote: undefined,
        votes: [],
    };


    voteSort (a, b) {
        return b.vote.localeCompare(a.vote);
    }

    render() {
        console.log(this.props.votes);
        return (
            <ul>
                {this.props.votes.slice().sort(this.voteSort).map(vote => (
                    <li key={`${vote.id}`}>
                        <Congressman congressman={vote.congressman} party={vote.congressman.party}>
                            <p><strong>{vote.vote}</strong></p>
                        </Congressman>
                    </li>
                ))}
            </ul>
        )
    }
}
