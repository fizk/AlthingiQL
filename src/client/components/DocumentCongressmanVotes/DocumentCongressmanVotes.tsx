import * as React from "react";
import Congressman from "../../elements/Congressman";

interface Picture {
    templateSrc: string
}

interface Party {
    id: number
    name: string
    color: string
}
type DocumentVote = {
    id: number
    voteId: number
    vote: string
    congressman: {
        id: number
        name: string
        avatar: Picture
        party: Party
    }
}

type DocumentCongressmanVotesProps = {
    assembly?: number,
    issue?: number,
    vote?: number,
    votes?: DocumentVote[]
};

export default class DocumentCongressmanVotes extends React.Component<DocumentCongressmanVotesProps, {}> {
    static defaultProps: DocumentCongressmanVotesProps = {
        assembly: undefined,
        issue: undefined,
        vote: undefined,
        votes: []
    };

    voteSort(a: {vote: string}, b: {vote: string}) {
        return b.vote.localeCompare(a.vote);
    }

    render() {
        return (
            <ul>
                {this.props.votes
                    .slice()
                    .sort(this.voteSort)
                    .map(vote => (
                        <li key={`${vote.id}`}>
                            <Congressman
                                congressman={vote.congressman}
                                party={vote.congressman.party}
                            >
                                <p>
                                    <strong>{vote.vote}</strong>
                                </p>
                            </Congressman>
                        </li>
                    ))}
            </ul>
        );
    }
}
