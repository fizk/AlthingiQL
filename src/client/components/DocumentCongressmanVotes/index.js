import {graphql, compose, gql} from 'react-apollo';
import DocumentCongressmanVotes from './DocumentCongressmanVotes';

const documentVotesQuery = gql`
    query documentVotes($assembly: Int! $issue: Int! $vote: Int!) {
        DocumentVotes(assembly: $assembly issue: $issue vote: $vote) {
            id
            voteId
            vote
            congressman {
                id
                name
                party {
                    id
                    name
                    color
                }
            }
        }
    }
`;

export default compose(
    graphql(documentVotesQuery, {
        props: all => ({
            votes: all.data.loading === false ? all.data.DocumentVotes : undefined,
            loading: all.data.loading,
        }),
        options: ({assembly, issue, vote}) => ({
            variables: {
                assembly: assembly,
                issue: issue,
                vote: vote,
            }
        })
    })
)(DocumentCongressmanVotes);

