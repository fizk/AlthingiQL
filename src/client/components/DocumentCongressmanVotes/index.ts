import {graphql, compose, gql} from 'react-apollo';
import DocumentCongressmanVotes from './DocumentCongressmanVotes';

type Picture = {
    templateSrc: string
}

type Party = {
    id: number
    name: string
    color: string
}

type Congressman = {
    id: number
    name: string
    avatar: Picture
    party: Party
}

type DocumentVote = {
    id: number
    voteId: number
    vote: string
    congressman: Congressman
}

const documentVotesQuery = gql`
    query documentVotes($assembly: Int! $issue: Int! $vote: Int!) {
        DocumentVotes(assembly: $assembly issue: $issue vote: $vote) {
            id
            voteId
            vote
            congressman {
                id
                name
                avatar {templateSrc}
                party {
                    id
                    name
                    color
                }
            }
        }
    }
`;

export default compose<any>( //@todo `any`
    graphql(documentVotesQuery, {
        props: (all: {data?: {loading: boolean, DocumentVotes: DocumentVote}}) => ({
            votes: all.data.loading === false ? all.data.DocumentVotes : undefined,
            loading: all.data.loading,
        }),
        options: ({assembly, issue, vote}: {assembly: number, issue: number, vote: number}) => ({
            variables: {
                assembly: assembly,
                issue: issue,
                vote: vote,
            }
        })
    })
)(DocumentCongressmanVotes);

