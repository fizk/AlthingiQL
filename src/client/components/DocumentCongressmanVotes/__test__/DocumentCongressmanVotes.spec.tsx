import * as React from 'react';
import DocumentCongressmanVotes from '../DocumentCongressmanVotes';
import renderer from 'react-test-renderer';
import {BrowserRouter as Router} from 'react-router-dom';
import {Vote} from '../../../../../@types';

type Partial<T> = { [P in keyof T]?: T[P]; };
type PartialVote = Partial<Vote>;

describe('<DocumentCongressmanVotes />', () => {
    test('default props', () => {
        const tree = renderer.create(
            <Router >
                <DocumentCongressmanVotes assembly={1} issue={2} vote={3} votes={[] as Vote[]} />
            </Router>,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    test('data', () => {
        const votes: PartialVote[] = [{
            id: 4,
            voteId: 5,
            vote: 'stuff',
            congressman: {
                id: 6,
                name: 'congressman',
                avatar: {
                    src: '',
                    templateSrc: '/path/to/file',
                },
                party: {
                    id: 7,
                    name: 'party',
                    color: '123123',
                },
            },
        }];
        const tree = renderer.create(
            <Router >
                <DocumentCongressmanVotes assembly={1} issue={2} vote={3} votes={votes as Vote[]} />
            </Router>,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
