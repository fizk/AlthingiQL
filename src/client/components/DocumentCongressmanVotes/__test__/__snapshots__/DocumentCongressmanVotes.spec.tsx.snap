// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DocumentCongressmanVotes /> data 1`] = `
<ul>
  <li>
    <article
      className="congressman"
    >
      <header
        className="congressman__avatar"
      >
        <div
          className="avatar"
          role="image"
          style={
            Object {
              "backgroundImage": "url(/path/to/file)",
            }
          }
          title="LjÃ³smynd af congressman"
        />
      </header>
      <div
        className="congressman__party"
      >
        <div
          className="badge"
          style={
            Object {
              "backgroundColor": "#123123",
            }
          }
          title="party"
        />
      </div>
      <div
        className="congressman__body"
      >
        <h4
          className="headline-3 headline-3--ellipsis"
        >
          congressman
        </h4>
        <p>
          <strong>
            stuff
          </strong>
        </p>
      </div>
    </article>
  </li>
</ul>
`;

exports[`<DocumentCongressmanVotes /> default props 1`] = `<ul />`;
